src/lib/encoder.ts

```diff
@@ -309,38 +311,38 @@ const decodeStringCompactBytesArray = (compactBytesArray: string): string[] => {
 
 const valueTypeEncodingMap = {
   bool: {
-    encode: (value: boolean) => abiCoder.encodeParameter('bool', value),
-    decode: (value: string) => abiCoder.decodeParameter('bool', value),
+    encode: (value: boolean) => (value === true ? '0x01' : '0x00'),
+    decode: (value: string) => value === '0x01',
   },
   boolean: {
-    encode: (value: boolean) => abiCoder.encodeParameter('bool', value),
-    decode: (value: string) => abiCoder.decodeParameter('bool', value),
+    encode: (value: boolean) => (value === true ? '0x01' : '0x00'),
+    decode: (value: string) => value === '0x01',
   },
   string: {
-    encode: (value: string) => abiCoder.encodeParameter('string', value),
-    decode: (value: string) => abiCoder.decodeParameter('string', value),
+    encode: (value: string) => utf8ToHex(value),
+    decode: (value: string) => hexToUtf8(value),
   },
   address: {
-    encode: (value: string) => abiCoder.encodeParameter('address', value),
-    decode: (value: string) => abiCoder.decodeParameter('address', value),
+    // TODO: if there is more than 20 bytes, truncate, if there is less than 20 bytes, throw an error
```

Comment:
how can an address have more than 20 bytes?

I would not truncate, i would throw an error.

Truncate would make us think yeah it works all good while using another "address string". Let's make it obvious that if length is not 20 then we throw

Reply:
@CJ42  I think we should throw. Nothing should be uncertain here.
If you give wrong input, we tell you wrong input. We don't try to make your wrong input work.

Reply:
Yes if not address, we throw. We can use `isAddress` from web3: https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#isaddress